sympy/matrices/matrices.py

```diff
@@ -2673,7 +2673,7 @@ def diagonal_solve(self, rhs):
         QRsolve
         pinv_solve
         """
-        if not self.is_diagonal:
+        if not self.is_diagonal():
             raise TypeError("Matrix should be diagonal")
         if rhs.rows != self.rows:
             raise TypeError("Size mis-match")
```

Comment:
Is there a test that hits this line?

Reply:
It already hits, if green line means the hit

https://codecov.io/gh/sympy/sympy/compare/f06edf42b003540cd119371440f6c226a63e5c8d...42acc42cbe890d7d3f99a5d313fc6ca8c7fa3b2c/diff

Reply:
I'm seeing green for 2676 and 2677. 2678 is yellow (partial) and 2679 is red.

Reply:
The coverage is not a duty from this commit though.
But it would be easy to cover this.
We may wait, but whatever, merging can be fine.